dwcj-engine/src/main/java/org/dwcj/concern/HasProperty.java

```diff
@@ -17,21 +19,48 @@
  */
 public interface HasProperty<T extends Component> {
 
+  /**
+   * Sets a property value.
+   *
+   * @param property the key of the property
+   * @param value the value to be set
+   *
+   * @return the component itself.
+   */
+  public T setProperty(String property, Object value);
+
   /**
    * Retrieves the value of the given property.
    *
+   * @param <V> the type of the property value returned
    * @param property the key of the property
+   * @param typeOfT the type of the property value returned to convert the value to.
```

Comment:
Is this saying that typeOfT is the type that you want the return value to be converted to? If so, that's maybe what I would rephrase this to say (I didn't do it just in case that isn't the meaning).

Reply:
Maybe it should be something like: `the type of the property value returned to cast the value to.` ?!  The method will attempt to convert the value returned by the client to the appropriate Java data type.  However, that this conversion depends on various factors, and although it usually works, there could be cases where it doesn't. So, providing a hint for the data type in such situations is necessary. Can you come with a better explanation